packages/eslint-plugin/docs/rules/restrict-template-expressions.md

```diff
@@ -0,0 +1,69 @@
+# Enforce template literal expressions to be of string type. (restrict-template-expressions)
+
+Examples of **correct** code:
+
+```ts
+const arg = 'foo';
+const msg1 = `arg = ${arg}`;
+const msg2 = `arg = ${arg || 'default'}`;
+```
+
+Examples of **incorrect** code:
+
+```ts
+const arg1 = [1, 2];
+const msg1 = `arg = ${arg}`;
+
+const arg2 = { name: 'Foo' };
+const msg2 = `arg = ${arg || null}`;
+```
+
+## Options
+
+The rule accepts an options object with the following properties:
+
+```ts
+type Options = {
+  // if true, also allow number type in template expressions
+  allowNumber?: boolean;
+  // if true, also allow boolean type in template expressions
+  allowBoolean?: boolean;
+  // if true, also allow null and undefined in template expressions
+  allowNullable?: boolean;
+};
+
+const defaults = {
+  allowNumber: false,
+  allowBoolean: false,
```

Comment:
IMO `allowNumber` and `allowBoolean` should be enabled by default. Numbers and booleans have completely reasonable string representation, so I don't see any advantages in forcing usage of `String(...)` for them

Reply:
I disagree. In my experience the default number representation is almost never what I want. Most commonly I use `n.toFixed(n)` so I don't get `0.300000000001` accidentally. I sometimes also want base other than 10. I also often want to have thousand separators. On frontend you should always use an i18n library so that period becomes a comma in certain languages. etc